src/main/java/com/williamfiset/algorithms/graphtheory/networkflow/examples/FordFulkersonExample.java

```diff
@@ -100,7 +100,6 @@ public NetworkFlowSolverBase(int n, int s, int t) {
 
     // Constructs an empty graph with n nodes including s and t.
     private void initializeEmptyFlowGraph() {
-      graph = new List[n];
```

Comment:
The graph variable cannot be deleted, it is being used in the next line. It may be causing warnings, but using a List[n] instead of a List<Edge> is a significant optimization for large networks. I would prefer to suppress that warnings for this class if they're annoying you. I don't know if that's done through the code or via Gradle.

Reply:
No, I am not deleting the variable. The variable is declared here:

  // The adjacency list representing the flow graph.
    protected List<Edge>[] graph;

I am deleting an instance, which is unnecessary because in the for loop is is replaced by another instance

in both cases, you use the "new" keyword
private void initializeEmptyFlowGraph() {
      **graph** = **new** List[n];
      for (int i = 0; i < n; i++) **graph**[i] = **new** ArrayList<Edge>();
    }

Reply:
`graph = new List[n];` allocates memory for the graph array. You can't index into graph[i] without getting a NPE.